testrepository/_computecontext.py

```diff
@@ -0,0 +1,88 @@
+#
+# Copyright (c) 2015 Testrepository Contributors
+#
+# Licensed under either the Apache License, Version 2.0 or the BSD 3-clause
+# license at the users choice. A copy of both licenses are available in the
+# project source as Apache-2.0 and BSD. You may not use this file except in
+# compliance with one of these two licences.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under these licenses is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# license you chose for the specific language governing permissions and
+# limitations under that license.
+
+"""Abstracts the idea of a place to run computations."""
+
+__all__ = ['Cache', 'Instance']
+
+import functools
+from collections import defaultdict, namedtuple
+from itertools import chain
+
+import six
+
+_Instance = namedtuple('Instance', ['profile', 'id'])
```

Comment:
I would strongly recommend using `pyrsistent.PClass` instead of namedtuple. It's not accidentally iterable, and it's much easier to encode types.

This class would become:

```python
class Instance(PClass):
  profile = field(text_type)
  id = field(text_type)
```